base/applications/cmdutils/tasklist/tasklist.c

```diff
@@ -0,0 +1,349 @@
+/*
+ * PROJECT:     ReactOS Tasklist Command
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Displays a list of currently running processes on the computer.
+ * COPYRIGHT:   Copyright 2020 He Yang (1160386205@qq.com)
+ */
+
+#include "tasklist.h"
+
+
+// the strings in opList are the command-line options.
+// should always correspond with the defines below, in sequence (except OP_PARAM_INVALID)
+static PCWSTR opList[] = { L"?", L"nh" };
+
+#define OP_PARAM_INVALID    -1
+#define OP_PARAM_HELP       0
+#define OP_PARAM_NOHEADER    1
+
+// the max string length PrintResString can handle
+#define RES_STR_MAXLEN 64
+
+// Print split line
+VOID PrintSplitLine(INT Length)
+{
+    for (INT i = 0; i < Length; i++)
+    {
+        ConPrintf(StdOut, L"=");
+    }
+}
+
+// Print spaces
+VOID PrintSpace(INT Length)
+{
+    for (; Length; Length--)
+    {
+        ConPrintf(StdOut, L" ");
+    }
```

Comment:
Note that only when filling using spaces, the shorter printf expression works too:
```suggestion
    ConPrintf(StdOut, L"%*s", Length, L"");
```

Reply:
Brilliant! I'm just thinking about is calling ```ConPrintf``` ```n``` times too performance costing

Reply:
yep! ConPuts in loop is less costly; however that trick I mentioned only works when you fill with spaces. For any other characters a loop of some sort is necessary.